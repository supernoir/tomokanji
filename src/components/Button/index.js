import React from 'react';

const Button = props => {
	return (
		<a onClick={props.handleClick} className={props.className}>
			{props.label}
		</a>
	);
};
export default Button;
