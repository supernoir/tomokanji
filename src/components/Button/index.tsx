import React from 'react';

interface ButtonProps {
	handleClick: () => void;
	label: string;
	className?: string;
}

const Button: React.FC<ButtonProps> = props => {
	return (
		<a onClick={props.handleClick} className={props.className}>
			{props.label}
		</a>
	);
};

export default Button