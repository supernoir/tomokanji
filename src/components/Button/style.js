import styled from 'styled-components';
import { Button } from './index';

export const StyledButton = styled(Button)`
  background: ${props => props.theme.white};
  color: ${props => props.theme.primary};
  min-width: 120px;
  font-weight: 400;
  font-size: 1.2rem;
  padding: .4em .8em;
  margin: .8em 0.6em;
  border-radius: .2em;
  cursor: pointer;
  text-decoration: none;
  transition: background 0.4s ease;
  box-shadow: 0px 4px 2px ${props => props.theme.primary};

  &:hover {
    color: ${props => props.theme.secondary};
    box-shadow: 0px 4px 2px ${props => props.theme.secondary};
    transition: box-shadow 0.4s ease-in-out;
  }
`;

export const StyledRandomButton = styled(Button)`

  background: ${props => `linear-gradient(135deg, ${props.theme.secondary}, ${props.theme.primary})`};
  color: ${props => props.theme.white};
  border: 3px solid ${props => props.theme.white};
  min-width: 120px;
  font-weight: 400;
  font-size: 1.2rem;
  padding: .4em .8em;
  margin: .8em 0.6em;
  border-radius: .2em;
  cursor: pointer;
  text-decoration: none;
  transition: background 0.4s ease;
  box-shadow: 0px 4px 2px ${props => props.theme.primary};


    &:hover {
      background: ${props => `linear-gradient(-135deg, ${props.theme.secondary}, ${props.theme.primary})`};
      box-shadow: 0px 4px 2px ${props => props.theme.secondary};
      transition:  box-shadow 0.4s ease-in-out;
      animation-name: shake;
      animation-duration: 80ms;
      animation-iteration-count: infinite;
      animation-direction: alternate
    }
    @keyframes shake {
      from {
        transform: rotate(-2deg);
      }
      to {
        transform: rotate(2deg);
      }
      
    }
`;
