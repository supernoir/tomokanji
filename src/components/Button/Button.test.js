import React from 'react';
import Button from './index';
import { render, shallow } from 'enzyme';

describe('Button', () => {
	//const testClickHandler = jest.fn();
	test('should match the snapshot', () => {
		const component = render(<Button label={'example'} />);
		expect(component).toMatchSnapshot();
	});

	test('should have a label', () => {
		const component = shallow(<Button label={'example'} />);
		expect(component.text('example')).toMatch('example');
	});
});
