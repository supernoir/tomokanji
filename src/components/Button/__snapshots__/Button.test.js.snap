// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<a>
  example
</a>
`;
